File: frontend/lib/api/audit.ts
// ── Audit API calls ──
import api from "./client";
import type { AuditLogEntry, AuditStats } from "@/lib/types";

interface AuditLogsParams {
    page?: number;
    per_page?: number;
    action_type?: string;
    status?: string;
}

export async function getAuditLogs(
    params: AuditLogsParams = {}
): Promise<{ items: AuditLogEntry[]; total: number; page: number; per_page: number }> {
    const { data } = await api.get("/audit/logs", { params });
    return data;
}

export async function getAuditStats(): Promise<AuditStats> {
    const { data } = await api.get<AuditStats>("/audit/stats");
    return data;
}

export async function exportAuditLogs(format: "json" | "csv" = "json"): Promise<{
    format: string;
    download_url: string | null;
}> {
    const { data } = await api.get("/audit/export", { params: { format } });
    return data;
}

File: backend/app/api/v1/audit.py
"""
Zia AI — Audit API
GET /logs, /export, /stats
"""

from fastapi import APIRouter, Depends
from app.api.deps import get_current_user

router = APIRouter()


@router.get("/logs")
async def get_logs(
    page: int = 1,
    per_page: int = 50,
    action_type: str = None,
    status: str = None,
    user: dict = Depends(get_current_user),
):
    """Get paginated audit logs with optional filters."""
    return {"items": [], "total": 0, "page": page, "per_page": per_page}


@router.get("/export")
async def export_logs(
    format: str = "json",
    user: dict = Depends(get_current_user),
):
    """Export audit logs as JSON or CSV."""
    return {"format": format, "message": "Export queued", "download_url": None}


@router.get("/stats")
async def get_stats(user: dict = Depends(get_current_user)):
    """Get action statistics (counts, trends, risk distribution)."""
    return {
        "total_actions": 0,
        "actions_today": 0,
        "success_rate": 0.0,
        "most_used_actions": {},
        "risk_distribution": {"low": 0, "medium": 0, "high": 0, "critical": 0},
    }
